sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -732,24 +735,25 @@ static pi_result getOrCreatePlatform(ze_driver_handle_t ZeDriver,
     auto VersionMajor = std::to_string((ZeDriverVersion & 0xFF000000) >> 24);
     auto VersionMinor = std::to_string((ZeDriverVersion & 0x00FF0000) >> 16);
     auto VersionBuild = std::to_string(ZeDriverVersion & 0x0000FFFF);
-    Platform[0]->ZeDriverVersion =
+    Platform->ZeDriverVersion =
         VersionMajor + "." + VersionMinor + "." + VersionBuild;
 
     ze_api_version_t ZeApiVersion;
     ZE_CALL(zeDriverGetApiVersion(ZeDriver, &ZeApiVersion));
-    Platform[0]->ZeDriverApiVersion =
+    Platform->ZeDriverApiVersion =
         std::to_string(ZE_MAJOR_VERSION(ZeApiVersion)) + "." +
         std::to_string(ZE_MINOR_VERSION(ZeApiVersion));
 
-    Platform[0]->ZeMaxCommandListCache = CommandListCacheSizeValue;
-    // save a copy in the cache for future uses.
-    PiPlatformsCache->push_back(Platform[0]);
+    Platform->ZeMaxCommandListCache = CommandListCacheSizeValue;
+    // Save a copy in the cache for future uses.
+    PiPlatformsCache->push_back(Platform);
+    // Copy the into cache to the out parameter.
+    *PlatformCache = PiPlatformsCache;
```

Comment:
Personally, I would move this line below to just before the return, so it's more clear that every return of `PI_SUCCESS` is paired with code that set `*PlatformCache`.  That's more of a stylistic thing, though.

Reply:
Just as a reminder the PiPlatformsCache is confined within the scope of the try block. Is this worth pulling the declaration of this vector out of the try block? If you think so than I will make this change. The other option could be to move the return statement up to the end of the try block, just after *PlatformCache is set.

Reply:
Ah, good point.  I take my comment back then.  It's probably better to leave the code as it is rather than moving the return.

Reply:
Done.